JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -1,211 +1,282 @@
-//
-//
-//        Project project = new Project();
-//        Hash hashes = new Hash();
-//
-//        //get group, corresponding to this id
-//        group = getGroup(groupId);
-//
-//        //-----------------------------------------set information related to project
-//        project.setProjectName(projectName);
-//        project.setGroup(group);
-//
-//        em.getTransaction().begin();
-//        em.persist(project);
-//        em.getTransaction().commit();
-//
-//        project = getProject(projectName, groupId);
-//
-//        //------------------------------------------set information related to hashes
-//        //  String projectHash = getHash(projectName);
-//        hashes.setProject(project);
-//        // hashes.setHash(projectHash);
-//
-//        em.getTransaction().begin();
-//        em.persist(hashes);
-//        em.getTransaction().commit();
-
-
-
-
-        return null;
+        return h;
 
     }
 
-    public void renameProject(String projectName, int groupId) {
+    private void updateHashFromBase(int projectId, Project p) throws UnsupportedEncodingException, NoSuchAlgorithmException {
 
-        Project project;
-        project = getProject(projectName, groupId);
+        String Hash = getHash(projectId);
 
-        project.setProjectName(projectName);
+        Hash h = (Hash)em.createQuery("SELECT h FROM Hash h WHERE h.hash=:hash")
+                .setParameter("hash", Hash)
+                .getSingleResult();
+
+        h.setProject(p);
+        em.persist(h);
 
-        em.getTransaction().begin();
-        em.persist(project);
-        em.getTransaction().commit();
     }
 
-    public void deleteProject(String projectName, int groupId) {
 
+    private Group getGroup(String groupName) {
 
-        Project project = getProject(projectName, groupId);
+        Group group;
+        try {
+            group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        } catch (NoResultException noResult) {
 
-        em.getTransaction().begin();
-        em.remove(project);
-        em.getTransaction().commit();
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
+
+        return group;
     }
 
+    private Group getGroup(int groupId) {
+
+        Group group;
+        try {
+            group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+                    .setParameter("groupid", groupId)
+                    .getSingleResult();
+        } catch (NoResultException noResult) {
 
+            logger.log(Level.WARNING, "NO RESULT IN QUERY", noResult);
+            return null;
+        }
 
-    public Project getProject(String projectName, int groupId) {
+        return group;
+    }
+
+
+
+
+    public String getHash(int projectId) throws NoSuchAlgorithmException, UnsupportedEncodingException {
```

Comment:
Dark magic. Also this is a bad way for generating hashCode of an entity.
Please search for best practices in the internet.